pkg/antctl/raw/set/flowaggregator/command_test.go

```diff
@@ -0,0 +1,294 @@
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			fakeKubeconfig, err := os.CreateTemp("", "fakeKubeconfig")
+			if err != nil {
+				log.Fatal(err)
+			}
+			defer os.Remove(fakeKubeconfig.Name())
+			fakeKubeconfig.Write(tc.fakeConfigs)
+			if cmd.Flag("kubeconfig") != nil {
+				cmd.Flag("kubeconfig").Value.Set(fakeKubeconfig.Name())
+			} else {
+				cmd.Flags().String("kubeconfig", fakeKubeconfig.Name(), "path of kubeconfig")
+			}
+			_, err = getk8sClient(cmd)
+			if tc.expectedErr == "" {
+				assert.NoError(t, err)
+			} else {
+				assert.Contains(t, err.Error(), tc.expectedErr)
+			}
+		})
+	}
+}
+
+func TestUpdateRunE(t *testing.T) {
+	cmd := NewFlowAggregatorSetCommand()
+	fakeConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testConfigMap",
+			Namespace: "flow-aggregator",
+		},
+	}
+
+	k8sClient := fake.NewSimpleClientset(fakeConfigMap)
+	tcs := []struct {
+		name           string
+		configMapName  string
+		podNamespace   string
+		args           []string
+		expectedErr    string
+		expectedOutput map[string]string
+	}{
+		{
+			name:           "valid ConfigMap and nil args",
+			configMapName:  "testConfigMap",
+			podNamespace:   "flow-aggregator",
+			expectedOutput: map[string]string{"flow-aggregator.conf": "{}\n"},
+		},
+		{
+			name:          "invalid ConfigMap and nil args",
+			configMapName: "",
+			podNamespace:  "flow-aggregator",
+			expectedErr:   "failed to locate flow-aggregator-config ConfigMap volume",
+		},
+		{
+			name:           "valid ConfigMap and valid args",
+			configMapName:  "testConfigMap",
+			podNamespace:   "flow-aggregator",
+			args:           []string{"clickHouse.enable=true"},
+			expectedOutput: map[string]string{"flow-aggregator.conf": "clickHouse:\n  enable: true\n"},
+		},
+		{
+			name:          "valid ConfigMap and invalid args",
+			configMapName: "testConfigMap",
+			podNamespace:  "flow-aggregator",
+			args:          []string{"clickhouse=true"},
+			expectedErr:   "unknown configuration parameter, please check antctl set flow-aggregator -h",
+		},
+		{
+			name:          "valid ConfigMap and invalid args syntax",
+			configMapName: "testConfigMap",
+			podNamespace:  "flow-aggregator",
+			args:          []string{"clickhouse.enable=true clickhopuse.debug=true"},
+			expectedErr:   "query should contain exactly one '='",
+		},
+		{
+			name:          "wrong Pod Namespace",
+			configMapName: "testConfigMap",
+			podNamespace:  "flow-visibility",
+			expectedErr:   "request namespace does not match object namespace",
+		},
+	}
+
+	getClients = func(cmd *cobra.Command) (kubernetes.Interface, error) {
+		return k8sClient, nil
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			os.Setenv("POD_NAMESPACE", tc.podNamespace)
+			os.Setenv("FA_CONFIG_MAP_NAME", tc.configMapName)
+			defer os.Unsetenv("POD_NAMESPACE")
+			defer os.Unsetenv("FA_CONFIG_MAP_NAME")
+			err := updateRunE(cmd, tc.args)
+			if err != nil {
```

Comment:
It should determine what check to do based on expectedErr, instead of the actual error, otherwise it would fail to detect real issue.
For example, when expectedErr is empty while actual err is not nil, the test would still succeed.